### DISCUSSION TITLE
Blocky Expected Performance

### DISCUSSION THREAD

**Discussion by sethkor** [0xERR0R/blocky#1657] [Q&A]
Hello I'm just wondering if anyone has some baseline performance stats for Blocky?  It seems to consume more CPU than I expect under load.  The example I have is under 500 QPS it's consuming 10% CPU on 1 16 core box.  I did see some discussion about number of inbound connections might be a concern but was wondering if there are any other metrics

Thanks in advance.

**Comment by Cyberlowspecs** [comment]
In my experience blocky is faster and more resource efficient than pihole. I have a VPN server that's being used by so many people and a feature of my VPN service is adblock. When i was using pihole, it consumed 70% of my CPU resources. People started complaining so i started researching for alternatives and landed on blocky. For the same number of clients blocky handles the load with less than 20% CPU usage
